waku/v2/waku_discv5.nim

```diff
@@ -77,9 +107,16 @@ proc new*(T: type WakuDiscoveryV5,
           flags: CapabilitiesBitfield,
           multiaddrs = newSeq[MultiAddress](),
           rng: ref HmacDrbgContext,
-          discv5Config: protocol.DiscoveryConfig = protocol.defaultDiscoveryConfig): T {.
+          topics: seq[string],
+          discv5Config: protocol.DiscoveryConfig = protocol.defaultDiscoveryConfig
+          ): T {.
   deprecated: "use the config and record proc variant instead".}=
 
+  let relayShardsRes = topicsToRelayShards(topics)
+
+  if relayShardsRes.isErr():
+    debug "pubsub topic parsing error", reason = relayShardsRes.error
```

Comment:
May be possible to skip one conditional check later by getting the `shards` option here already.
Something like:
```suggestion
  var relayShardsOp: Option[RelayShard]
  if relayShardsRes.isOk():
    relayShardsOp = relayShardsRes.get()
  else
    debug "pubsub topic parsing error", reason = relayShardsRes.error
    relayShardsOp = none(RelayShard)
```

That way the `record` block below can be a bit cleaner by just checking the option before modifying the builder. Wdyt?

Reply:
Agree